docs/DeveloperGuide.md

```diff
@@ -1455,4 +1456,9 @@ Implementation of the `Lesson` features took ~6-7k LOC, including tests.
 
 **Fee Calculator**
 
+**Viewing tags feature**
 
+There are multiple ways of implementing this enhancement. The simplest way would be to just create a method in `Addressbook` to add `Tag` objects from each `Person` in `UniquePersonList` whenever users request to view the list of tags.
+However, to abide by the Separation of Concerns (SoC) principle, the `UniqueTagList` class is created for adding distinct tags from `Person` into the tag list. 
+One of the alternative implementations is to keep a list of `Person` objects with the same `Tag` in the `Tag` object to keep track of the number of students labelled with that `Tag`. However, this could introduce circular dependencies.
```

Comment:
```suggestion
One of the alternative implementations is to keep a list of `Person` objects with the same `Tag` in the `Tag` object to keep track of the number of students labeled with that `Tag`. However, this could introduce circular dependencies.
```